stubs/Reflection.phpstub

```diff
@@ -41,23 +222,205 @@ class ReflectionClass implements Reflector {
      */
     public function newInstanceWithoutConstructor(): object {}
 
+    /** @psalm-pure */
+    public function getParentClass(): ?ReflectionClass {}
+
+    /**
+     * @param ReflectionClass|class-string $class
+     *
+     * @psalm-pure
+     */
+    public function isSubclassOf($class): bool {}
+
+    /** @return array<non-empty-string, mixed> */
+    public function getStaticProperties(): array {}
+
+    /**
+     * @return array<non-empty-string, mixed>
+     *
+     * @psalm-pure
+     */
+    public function getDefaultProperties(): array {}
+
+    /** @psalm-pure */
+    public function isIterateable(): bool {}
+
+    /** @psalm-pure */
+    public function isIterable(): bool {}
+
+    /**
+     * @param self|class-string $interface
+     *
+     * @psalm-pure
+     */
+    public function implementsInterface($interface): bool {}
+
+    /** @psalm-pure */
+    public function getExtension(): ?ReflectionExtension {}
+
+    /**
+     * @return non-empty-string|false
+     *
+     * @psalm-pure
+     */
+    public function getExtensionName() {}
+
+    /** @psalm-pure */
+    public function inNamespace(): bool {}
+
+    /** @psalm-pure */
+    public function getNamespaceName(): string {}
+
+    /**
+     * @return non-empty-string
+     *
+     * @psalm-pure
+     */
+    public function getShortName(): string {}
+
     /**
      * @return ?array<string>
      * @psalm-ignore-nullable-return
+     * @psalm-pure
      */
     public function getTraitNames(): array {}
 
     /**
      * @since 8.0
      * @template TClass as object
      * @param class-string<TClass>|null $name
-     * @return ($name is null ? array<ReflectionAttribute<object>> : array<ReflectionAttribute<TClass>>)
+     * @return ($name is null ? list<ReflectionAttribute<object>> : list<ReflectionAttribute<TClass>>)
+     * @psalm-pure
      */
     public function getAttributes(?string $name = null, int $flags = 0): array {}
 }
 
-class ReflectionFunction implements Reflector
+abstract class ReflectionFunctionAbstract implements Reflector
 {
+    /** @psalm-pure */
+    public function inNamespace(): bool {}
+
+    /** @psalm-pure */
+    public function isClosure(): bool {}
+
+    /** @psalm-pure */
+    public function isDeprecated(): bool {}
+
+    /** @psalm-pure */
+    public function isInternal(): bool {}
+
+    /** @psalm-pure */
+    public function isUserDefined(): bool {}
+
+    public function getClosureThis(): ?object {}
+
+    /** @psalm-pure */
+    public function getClosureScopeClass(): ?ReflectionClass {}
+
+    /** @psalm-pure */
+    public function getClosureCalledClass(): ?ReflectionClass {}
+
+    /**
+     * @return non-empty-string|false
+     *
+     * @psalm-pure
+     */
+    public function getDocComment() {}
```

Comment:
I experimented a bit with this: the minimal doc comment that can be written is `/** */`, and it will produce the string `'/** */'`.

It is not possible to get a `string&empty` doc comment from reflection